middleman-core/lib/middleman-core/preview_server/server_information.rb

```diff
@@ -0,0 +1,135 @@
+require 'socket'
+require 'active_support/core_ext/object/blank'
+require 'middleman-core/dns_resolver'
+require 'middleman-core/preview_server/validator'
+require 'middleman-core/preview_server/information'
+require 'middleman-core/preview_server/network_interface_inventory'
+
```

Comment:
Here's where most of the magic happens.

* It asks for name resolution
* It asks for validation (is missing for now)

It needs some cleanup though.